src/V3Randomize.cpp

```diff
@@ -1369,17 +1376,24 @@ class RandomizeVisitor final : public VNVisitor {
             return new AstForeach{fl, randLoopVarp, newRandStmtsp(fl, tempElementp, nullptr)};
         };
         AstNodeExpr* tempElementp = nullptr;
-        while (VN_CAST(tempDTypep, DynArrayDType) || VN_CAST(tempDTypep, UnpackArrayDType)) {
+        while (VN_CAST(tempDTypep, DynArrayDType) || VN_CAST(tempDTypep, UnpackArrayDType)
+               || VN_CAST(tempDTypep, AssocArrayDType) || VN_CAST(tempDTypep, QueueDType)) {
             AstVar* const newRandLoopIndxp = createLoopIndex(tempDTypep);
             randLoopIndxp = AstNode::addNext(randLoopIndxp, newRandLoopIndxp);
             tempElementp
-                = VN_CAST(tempDTypep, DynArrayDType)
-                      ? static_cast<AstNodeExpr*>(
-                          new AstCMethodHard{fl, tempElementp ? tempElementp : exprp, "atWrite",
-                                             new AstVarRef{fl, newRandLoopIndxp, VAccess::READ}})
-                      : static_cast<AstNodeExpr*>(
-                          new AstArraySel{fl, tempElementp ? tempElementp : exprp,
-                                          new AstVarRef{fl, newRandLoopIndxp, VAccess::READ}});
+                = VN_CAST(tempDTypep, DynArrayDType) ? static_cast<AstNodeExpr*>(
+                      new AstCMethodHard{fl, tempElementp ? tempElementp : exprp, "atWrite",
+                                         new AstVarRef{fl, newRandLoopIndxp, VAccess::READ}})
+                  : VN_CAST(tempDTypep, UnpackArrayDType) ? static_cast<AstNodeExpr*>(
+                        new AstArraySel{fl, tempElementp ? tempElementp : exprp,
+                                        new AstVarRef{fl, newRandLoopIndxp, VAccess::READ}})
+                  : VN_CAST(tempDTypep, AssocArrayDType) ? static_cast<AstNodeExpr*>(
+                        new AstAssocSel{fl, tempElementp ? tempElementp : exprp,
+                                        new AstVarRef{fl, newRandLoopIndxp, VAccess::READ}})
+                  : VN_CAST(tempDTypep, QueueDType) ? static_cast<AstNodeExpr*>(new AstCMethodHard{
+                        fl, tempElementp ? tempElementp : exprp, "atWriteAppend",
+                        new AstVarRef{fl, newRandLoopIndxp, VAccess::READ}})
+                                                    : nullptr;
```

Comment:
I think this would be clearer as a series of `if` and assignments.  Then the casts shouldn't be needed.